### PR TITLE
refactor rooch config and chainid to support multi env for rooch store, including devnet/testnet/mainnet

### PR DESCRIPTION
**Pull request by baichuan3**
resolver https://github.com/rooch-network/rooch/issues/685 and https://github.com/rooch-network/rooch/issues/689

Summary:

1. refactored chainid, support built-in ChainID (dev/test/main) and user-defined ChainID (id+name)

2. When the rooch server starts, it supports specifying chainid parameters, which are `-n dev`, `-n test`, `-n main`, which will generate db files with different paths according to the chainid, and start by `test network` by default;

3. In the `dev network`, the db will default use a temporary file directory. After each server shutdown, the db file will be automatically deleted; the `test network` and `main network` networks will retain the db file

4. Correspondingly, the cleanup command also needs to specify the `chainid`, which is still the `test network` network by default;

5. At the same time, the server startup error prompt is optimized. If the db schema is abnormal, the user is prompted to clean up the db file

### CODE REVIEW

**Review by jolestar — APPROVED** [review]

**Comment by baichuan3** [comment]
Encountered similar version errors several times, from different dependencies, related to the migration to google gke? @yourmoonlight @jolestar 

```
Run cargo test -p testsuite --test integration
error: package `anstream v0.5.0` cannot be built because it requires rustc 1.70.0 or newer, while the currently active rustc version is 1.[6](https://github.com/rooch-network/rooch/actions/runs/5975123766/job/16212745764?pr=701#step:10:7)8.0
Either upgrade to rustc 1.70.0 or newer, or use
cargo update -p anstream@0.5.0 --precise ver
where `ver` is the latest version of `anstream` supporting rustc 1.68.0
Error: Process completed with exit code [10](https://github.com/rooch-network/rooch/actions/runs/5975123766/job/16212745764?pr=701#step:10:11)1.
```

File: crates/rooch/src/commands/server/commands/start.rs [hunk 1]
```diff
@@ -14,20 +17,27 @@ use tracing::info;
...
 pub struct StartCommand {
+    // #[serde(skip_serializing_if = "Option::is_none")]
+    #[clap(long, short = 'n', help = R_OPT_NET_HELP)]
+    pub chain_id: Option<RoochChainID>,
+
     /// If true, start the service with a temporary data store.
     /// All data will be deleted when the service is stopped.
     #[clap(long, parse(from_flag))]
-    pub temp_db: bool,
+    pub is_temp_store: bool,
```
[the remark this answers is not in the document]
Comment (baichuan3) [reply crates/rooch/src/commands/server/commands/start.rs:27]: The `is_temp_store flag` has always existed,  so still exposed to the cli to all chain net.
If don’t need it later, I will remove it.

File: crates/rooch/src/commands/init.rs [hunk 2]
```diff
@@ -116,9 +116,9 @@ impl CommandAction<String> for Init {
                 .persisted(client_config_path.as_path())
                 .save()?;
 
-                //Store config init
-                StoreConfig::init()
-                    .map_err(|e| anyhow::anyhow!("Init stroe config failed:{}", e))?;
+                // //Store config init
+                // StoreConfig::init()
+                //     .map_err(|e| anyhow::anyhow!("Init stroe config failed:{}", e))?;
```
[the remark this answers is not in the document]
Comment (baichuan3) [reply crates/rooch/src/commands/init.rs]: > This should be removed?

already removed

**Comment by jolestar** [comment]
@baichuan3 upgrade rust to 1.70.0 or fix the clap to a particular minor version.

Maybe we need to add the `Cargo.lock` to the source code repo, to avoid the dependency auto update.

File: crates/rooch/src/commands/server/commands/start.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (baichuan3) [reply crates/rooch/src/commands/server/commands/start.rs]: removed

**Comment by baichuan3** [comment]
> @baichuan3 upgrade rust to 1.70.0 or lock the clap to a particular minor version.
> 
> Maybe we need to add the `Cargo.lock` to the source code repo, to avoid the dependency auto update.

Try to upgrade rust to lastest stable ver 1.72.0, there are some changes in syntax cause ci to fail, need another pr to handle it.

**Comment by jolestar** [comment]
We should remove `Cargo.lock` from .gitignore and add the `Cargo.lock` to git.

https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html

> If you’re building a non-end product, such as a rust library that other rust [packages](https://doc.rust-lang.org/cargo/appendix/glossary.html#package) will depend on, put Cargo.lock in your .gitignore. If you’re building an end product, which are executable like command-line tool or an application, or a system library with crate-type of staticlib or cdylib, check Cargo.lock into git. If you’re curious about why that is, see [“Why do binaries have Cargo.lock in version control, but not libraries?” in the FAQ](https://doc.rust-lang.org/cargo/faq.html#why-do-binaries-have-cargolock-in-version-control-but-not-libraries).


In the beginning, this repo is for MoveOS, a library, so we ignore the `Cargo.lock`.

But now, it is a binary repo for `rooch`, so we should add `Cargo.lock` to git.

**Pull request merged**